docs/tasks/debug-application-cluster/core-metrics-pipeline.md

```diff
@@ -0,0 +1,47 @@
+---
+approvers:
+- fgrzadkowski
+- piosz
+title: Core metrics pipeline
+---
+
+Resource usage metrics are important for:
+
+- Kubernetes components to operate (e.g. autoscalers to automatically scale user applications)
+- Kubernetes users to understand their applications behavior
+
+In Kubernetes they are available through Metrics API.
+
+## Metrics API
+
+Metrics API provides current resource usage metrics for pods and nodes. It doesn't offer historical metrics, so one needs to
+archive metrics themselves. In the future Historical API might be defined but it will not be implemented as a part of
+the core metrics pipeline.
+
+The API is discoverable through the same endpoint as the other Kubernetes APIs under `/apis/metrics.k8s.io/` path.
+It also offers the same security, scalability and reliability guarantees.
```

Comment:
What this paragraph is saying `this API is no different from any other API`. Is it worth mentioning explicitly?

Reply:
@crassirostris done

@tengqm The API was defined in alpha ~Q2'16, so quite some time ago and implemented in [Heapster](https://github.com/kubernetes/heapster/tree/master/metrics/apis/metrics). This was not the right way of implementing the API though, so in Q2'17 we [designed](https://docs.google.com/document/d/1w6-ZfnA18aKYLJ8DCLBFKlv_1umm74x0I2X4V188kvU/edit) an implemented metrics-server, which is the proper way of implementing the API. Then it was graduated to beta and metrics-server became enabled by default in Kubernetes in 1.8.

As the API is not implemented by API server it's requires metrics-server to be deployed to Kubernetes. As metric s-server was not deployed by default before v1.8 we haven't seen the API there.

Reply:
Thanks for the background. I just learned that a metrics-server have to be deployed with its API (APIService) registered to the kube-aggregator (or just kube-apiserver) for discovery.